data-prepper-plugins/date-processor/README.md

```diff
@@ -0,0 +1,115 @@
+      record_type: "event"
+      format: "json"
+  prepper:
+    - grok:
+        match: 
+          log: ['%{IPORHOST:clientip} \[%{HTTPDATE:timestamp}\] %{NUMBER:response_status:int}']
+          
+    - date:
+        match:
+            - key: timestamp
+              patterns: ["dd/MMM/yyyy:HH:mm:ss XX"] 
+        destination: "@timestamp"
+        timezone: "America/Chicago"
+        locale: "en_US"
+  sink:
+    - stdout:
+```
+Create the following file named `logs_json.log` and replace the `path` in the file source of your `pipeline.yaml` with the path of this file.
+
+```json
+{"message": "127.0.0.1 198.126.12 [10/Oct/2000:13:55:36 -0700] 200"}
+```
+
+The date processor configuration from `pipeline.yaml` will parse the timestamp key generated by grok processor by converting it to ISO 8601 format.
+
+When you run Data Prepper with this `pipeline.yaml` passed in, you should see the following standard output.
+```json
+{ 
+  "message":"127.0.0.1 198.126.12 [10/Oct/2000:13:55:36 -0700] 200",
+  "response_status":200,
+  "clientip":"198.126.12",
+  "timestamp":"10/Oct/2000:13:55:36 -0700",
+  "@timestamp": "2000-10-10'T'13:55:36.000-07:00"
+}
+```
+
+## Configuration
+
+* `from_time_received`: A boolean that is used for adding default timestamp to the event which is when Data Prepper source first receives 
+  the event. It takes the timestamp from event metadata and stores it in destination. In the absence of this option, Data Prepper doesn't 
+  add a default timestamp field to any event.
+  
+  * Type: boolean
+  * Default: false
+
+  The following example of date configuration will use timestamp when the event is first seen by Data Prepper source and uses it as a 
+  default timestamp for the event and stores it in `@timestamp`.
+```yaml
+prepper:
+  - date:
+      from_time_received: true
+      destination: "@timestamp"
+```
+
+* `match`: A list of `key` and `patterns` which specifies key of record to match patterns against. List can only have one with a 
+valid key and at least one pattern is required if match is configured.
+  * Type: List
+  * Default: no default value
+    * `key`: key of record to match patterns against.
+      * Type: String
+    * `patterns`: List of possible patterns the timestamp value of key can have.
+      * Type: `List<String>`
+
+The following example of date configuration will use `timestamp` key to match against given patterns and stores the timestamp in ISO 8601
+format in destination. 
+```yaml
+prepper:
+  - date:
+      match:
+        - key: timestamp
+          patterns: ["dd/MMM/yyyy:HH:mm:ss XX", "MM/dd/yyyy"]
+```
+> :warning: **`from_time_received` and `match` are mutually exclusive.** Either use `from_time_received` or `match` but not both. Use multiple date filters if 
+> both options should be used.
+* `destination` (Optional): Field to store the timestamp parsed by date processor. It can be used with both `match` and `from_time_received`.
+  * Type: String
+  * Default: `"@timestamp"`
```

Comment:
Consider adding documentation for why do we have `@` prefix for the default destination field.